// feel-ui/src/input.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
(function (factory) {
    if (typeof module === 'object' && typeof module.exports === 'object') {
        var v = factory(require, exports); if (v !== undefined) module.exports = v;
    }
    else if (typeof define === 'function' && define.amd) {
        define(["require", "exports", "angular2/core", "angular2/common"], factory);
    }
})(function (require, exports) {
    "use strict";
    var core_1 = require("angular2/core");
    var common_1 = require("angular2/common");
    var InputComponent = (function () {
        function InputComponent() {
            this.multiline = false;
            this.type = "text";
            this.valueChange = new core_1.EventEmitter();
            this._rows = 2;
        }
        InputComponent.prototype.onInput = function (event) {
            this.value = event;
            this.valueChange.emit(event);
            var rows = this.value.split("\n").length;
            if (rows < 2) {
                rows = 2;
            }
            this._rows = rows;
        };
        __decorate([
            core_1.Input(), 
            __metadata('design:type', String)
        ], InputComponent.prototype, "label", void 0);
        __decorate([
            core_1.Input(), 
            __metadata('design:type', String)
        ], InputComponent.prototype, "value", void 0);
        __decorate([
            core_1.Input(), 
            __metadata('design:type', Boolean)
        ], InputComponent.prototype, "multiline", void 0);
        __decorate([
            core_1.Input(), 
            __metadata('design:type', String)
        ], InputComponent.prototype, "type", void 0);
        __decorate([
            core_1.Output(), 
            __metadata('design:type', Object)
        ], InputComponent.prototype, "valueChange", void 0);
        InputComponent = __decorate([
            core_1.Component({
                directives: [common_1.NgClass, common_1.NgIf],
                selector: "feel-input",
                template: "<label>{{label}}</label>\n               <textarea *ngIf=\"multiline\" [ngModel]=value (ngModelChange)=\"onInput($event)\" [rows]=_rows></textarea>\n               <input *ngIf=\"!multiline\" [type]=type [ngModel]=value (ngModelChange)=\"onInput($event)\" />"
            }), 
            __metadata('design:paramtypes', [])
        ], InputComponent);
        return InputComponent;
    }());
    exports.InputComponent = InputComponent;
});
