// feel-ui/feel-ui.js
//Temporary work around for Angular2#4902
///<reference path="./node_modules/angular2/typings/browser.d.ts"/>
(function (factory) {
    if (typeof module === 'object' && typeof module.exports === 'object') {
        var v = factory(require, exports); if (v !== undefined) module.exports = v;
    }
    else if (typeof define === 'function' && define.amd) {
        define(["require", "exports", "./src/button", "./src/menu-bar", "./src/menu-panel", "./src/input", "./src/notification-provider", "./src/table", "./src/slider"], factory);
    }
})(function (require, exports) {
    var button_1 = require("./src/button");
    exports.ButtonComponent = button_1.ButtonComponent;
    var menu_bar_1 = require("./src/menu-bar");
    exports.MenuBar = menu_bar_1.MenuBar;
    var menu_panel_1 = require("./src/menu-panel");
    exports.MenuPanel = menu_panel_1.MenuPanel;
    var input_1 = require("./src/input");
    exports.InputComponent = input_1.InputComponent;
    var notification_provider_1 = require("./src/notification-provider");
    exports.NotificationProvider = notification_provider_1.NotificationProvider;
    var table_1 = require("./src/table");
    exports.TableComponent = table_1.TableComponent;
    var slider_1 = require("./src/slider");
    exports.Slider = slider_1.Slider;
});
